### PR TITLE
WIP - 59 layout

### PR DESCRIPTION
**Pull request by Ataww**

### CODE REVIEW

File: src/components/TheCanvas.vue [hunk 1]
```diff
@@ -7,139 +7,140 @@
...
-  tools: ITool[] = []
-  $refs!: {
-    layer: any
-  }
+    stageSize = {
+      width: window.innerWidth,
+      height: window.innerHeight
+    }
+    $refs!: {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bryansamuel1) [inline src/components/TheCanvas.vue]: isn't @ts-ignore sufficient here?

File: src/main.ts [hunk 2]
```diff
@@ -6,9 +6,6 @@ import './registerServiceWorker'
 import router from './router/index'
 import store from './store'
 
-let socket = new WebSocket('ws://localhost:80')
```
Comment (bryansamuel1) [inline src/main.ts]: where has this gone?

File: src/mixins/CanvasTools.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import Konva from 'konva'
```
Comment (bryansamuel1) [inline src/mixins/CanvasTools.ts]: what has changed in this file? or have you just moved it?

File: src/plugins/vuetify.ts [hunk 4]
```diff
@@ -40,7 +40,8 @@ Vue.use(Vuetify, {
   directives: { }
 })
 
-const opts:Object = {
```
Comment (bryansamuel1) [inline src/plugins/vuetify.ts]: opts?

File: src/store/index.ts [hunk 5]
```diff
@@ -1,87 +1,21 @@
...
-  enabledTool: string;
-  enabled: boolean;
-  tools: ITool[]
+export enum Namespaces {
+  TOOLS = 'tools',
+  CURSOR = 'cursor',
+  SOCKET = 'socket'
 }
 
-export default new Vuex.Store<IVuexState>({
```
Comment (bryansamuel1) [inline src/store/index.ts]: I am right in thinking you split this out into modules? anything else done?

File: src/components/TheCanvas.vue (hunk 1 above)
Comment (Ataww) [reply src/components/TheCanvas.vue]: I prefer using eslint disable as ts-ignore does completely bypass the ts compiler

File: src/components/TheCanvas.vue [hunk 6]
```diff
@@ -7,139 +7,140 @@
...
       }
-    })
-  }
+    }
 
-  onMouseUpHandler (e: any) : void {
-    this.$store.dispatch('setDisabled')
-  }
+    onMouseDownHandler (e: Konva.KonvaPointerEvent): void {
+      if (this.activeTool) {
```
[the remark this answers is not in the document]
Comment (Ataww) [reply src/components/TheCanvas.vue]: should be enabledTool, but it's just an alias to the currently active (selected) tool in the store

File: src/components/TheCanvas.vue [hunk 7]
```diff
@@ -7,139 +7,140 @@
...
-  }
 
-  onMouseMoveHandler (e: any) : void {
-    let vm: IVmThis = this
-    const enabledTool = vm.$data.enabledTool
-    if (enabledTool && vm.$data.enabled) {
-      vm[enabledTool]().onClick(e, vm.$refs.stage.getNode(), vm.$refs.layer.getNode(), this.$store.getters.tool(enabledTool))
-      vm.$socket.send(this.webSocketSendData(enabledTool, e))
+    onWebSocketHandler (e: Event): void {
+      this.activeTool.action(e as unknown as Konva.KonvaPointerEvent, this.stageNode, this.layerNode)
```
[the remark this answers is not in the document]
Comment (Ataww) [reply src/components/TheCanvas.vue]: yes I've hacked this together real quick

File: src/main.ts (hunk 2 above)
Comment (Ataww) [reply src/main.ts]: in the store, you have a socket module now

File: src/mixins/CanvasTools.ts (hunk 3 above)
Comment (Ataww) [reply src/mixins/CanvasTools.ts]: it's a class now, rather than an object. See https://github.com/vuejs/vue-class-component#using-mixins

**Pull request merged**